### DISCUSSION TITLE
Why do you do durexForth?

### DISCUSSION THREAD

**Discussion by Whammo** [jkotlinski/durexforth#416] [General]
I do durexForth because I've always enjoyed programming and gaining understanding of the C64.
 BASIC is slow and hard on the eyes, a full screen of it's code is too busy, repetitive and long lined.
Code for durexForth is as compact, easy on the eyes and powerful as you want it to be. Everything new that I do is like a world record.
Never before coded in durexForth!
 The truth is, in this day and age, I see my programming for the C64 as folk art. So rather than spinning pots or making little wooden windmills featuring a little man chopping wood, I do durexForth.
 I started using durexForth many years ago looking for a programming language that was fast, wasn't a pain to compile and interfaced well with assembly. I chose a piece of code, and ported it to as many language packages as I could manage. Promal was a close second.
 It's amazing how much more stable durexForth has become as my skills have improved.

**Comment by jkotlinski** [comment]
For me, I don't really use durexForth, I only work on the environment itself.
I started it a long time ago as a fun experiment, since I was attracted to the Forth philosophy and way of thinking.
For me, this is the perfect way to learn more and explore further.
I enjoyed C64 since the 80s, it's not so much of "nostalgy" but more something I was into since I was a kid. It is one of my native platforms.

**Comment by lonetech** [comment]
I was a little too late to be a C64 native, but was fascinated with computers since a young age. I started learning programming on a Casio calculator and STOS BASIC on Atari ST. Since learning about Forth via Open Firmware, I've come to recognize it as the type of language we ought to have had rather than BASIC; one that's less in the way of both execution and understanding. It wasn't my first stack oriented language, as I actually wrote some PostScript, but that's inspired by Forth anyway. To me, durexForth is one of the paths demonstrating Forth on real hardware, while I'm also looking at options like Mecrisp and ColorForth, and toying with Gforth, which turned out to be the easy option on Android. I still don't have a real C64, but am considering acquiring a Mega65, which would also be able to run e.g. J1a or microCore. 

In particular, durexForth's inclusion of the v editor, turnkey save-prg and start variable are strong features. This makes it not only a complete programming environment but one easily able to produce deployable programs, albeit with a somewhat large runtime. All without needing some separate computer, and with a large part of the source code included on disk too. I love the sort of learning opportunities granted by going from "include rnd" to "v rnd".

**Comment by ekipan** [comment]
I first saw mention of durexForth [1] about a month ago in the comments of a Youtube [video about 64 Forth](https://youtu.be/1XdgUK1NbpI). I've noodled with gforth a little bit but I was compelled to try out a web emulator and touch the C64 for the first time, eventually downloading VICE proper and saving programs to disk image. 

My childhood 6502 machine was an NES, and for the last week my head's been aflurry with ideas for a memory viewer/editor with a controller interface inspired by LSDJ. I've never programmed an NES either, so that's not really more than _what the screen would look like_ and _what the 8 buttons would do._ If you can edit memory you might be able to write and interpret forth. Nintendo wrote a BASIC after all.

There's a cross-compiling forth system [targetting the Game Boy](https://github.com/ams-hackers/gbforth), which has 4 times the NES's RAM (2k -> 8k), though that's a Z80 machine, and doesn't include the interpreter or compiler on target (because they didn't know how one would edit source, they say).

[1] is there an official styling? a lot of the source says "durexForth" but there's also "durexforth" and "DurexForth."

**Reply by jkotlinski** [reply]
Welcome! :-) 

I currently prefer "durexForth". "durexforth" is practical for filenames and URLs.
"DurexForth" may sometimes look good when it is the first word in a sentence? Not sure.

 durexFORTH is another interesting option that has not been tried at all yet.

**Reply by IanWitham** [reply]
I think the same YouTube comment led me to durexForth. I was reading about the Minstrel 4D, a kit form recreation of the Jupiter Ace, which was an 80s microcomputer with Forth in ROM instead of the ubiquitous BASIC. That got me interested enough to find Robin's video (which I must have missed when it came out), and now I'm here.

I use durexForth because it's a very direct and fun way to experiment with the C64 platform, and standard enough to learn about Forth in general. My next steps may be to obtain something like a Scamp3 microcontroller (a bit like an arduino or Pico which runs Forth), or figure out how to use something like GForth for creative coding.

Edit: Just to echo Johan... Coding durexForth on a C64 doesn't feel "retro" to me at all, because I use it to create something new. It's certainly a constrained environment which is part of the fun.
